api/src/services/sample-location-service.ts

```diff
@@ -52,6 +52,14 @@ export class SampleLocationService extends DBService {
    * @memberof SampleLocationService
    */
   async deleteSampleLocationRecord(surveySampleSiteId: number): Promise<SampleLocationRecord> {
+    const sampleMethodService = new SampleMethodService(this.connection);
+
+    // Delete all methods associated with the sample location
+    const existingSampleMethods = await sampleMethodService.getSampleMethodsForSurveySampleSiteId(surveySampleSiteId);
+    for (const item of existingSampleMethods) {
+      await sampleMethodService.deleteSampleMethodRecord(item.survey_sample_method_id);
```

Comment:
We should probably use a method that deletes all methods by an array of method IDs

Reply:
I found this was tricky to do as each method has the potential to have many periods associated to the method. The `deleteSampleMethodRecord` function handles the multi delete of each connected period to the method.

Reply:
Can we not just change `SampleLocationRepository::deleteSampleLocationRecord` accordingly?

```javascript
  /**
   * Deletes a survey Sample Location, including all of its sample methods and all of its sample methods'
   * sample periods.
   *
   * @param {number} surveySampleSiteId
   * @return {*}  {Promise<SampleLocationRecord>} an array of SampleLocationRecords that have been deleted
   * @memberof SampleLocationRepository
   */
async deleteSampleLocationRecord(surveySampleSiteId: number): Promise<SampleLocationRecord> {
    const sqlStatement = SQL`
      WITH site_method_ids AS (
        SELECT
          survey_sample_method_id
        FROM
          survey_sample_method
        WHERE
          survey_sample_site_id = ${surveySampleSiteId}
      ),
      delete_child_records AS (
        -- Delete sample periods --
        DELETE FROM
          survey_sample_period
        WHERE
          survey_sample_method_id IN (site_method_ids);
        -- Delete sample methods --
        DELETE FROM
          survey_sample_method
        WHERE
          survey_sample_method_id IN (site_method_ids);
      ) DELETE FROM
        survey_sample_site
      WHERE
        survey_sample_site_id = ${surveySampleSiteId}
      RETURNING
        *;
    `;
```

Reply:
Forgive me if you already pursued this an it didn't work